### ISSUE TITLE
Make tokenizer return information about case of the word

### ISSUE THREAD

**Issue by maiiabocharova** [huggingface/tokenizers#1232]
Can you please advice me on how I should modify the code to make tokenizer return information about was/was not the token capitalized? I wrote such functionality, but it slowed down tokenizer too much.

```
class CustomTokenizer:
    def __init__(self):
        self.tokenizer = AutoTokenizer.from_pretrained('bert-base-uncased')
        self.mask_token = tokenizer.mask_token
        self._pad_token = tokenizer._pad_token
        self.pad_token_id = tokenizer.pad_token_id
        self.padding_side = tokenizer.padding_side
    def __call__(
                self,
                text, # Union[TextInput, PreTokenizedInput, List[TextInput], List[PreTokenizedInput]]
                return_tensors='pt',
                padding='max_length',
                max_length=128,
            ):
        if isinstance(text, str):
            text = pre_tokenizer.pre_tokenize_str(text)
            input_ids, capitalization_ids = [], []
            for word in text:
                word_ids = self.tokenizer(
                    word[0].lower(), 
                    add_special_tokens=False
                )['input_ids']
                input_ids.extend(word_ids)
                if word[0].istitle():
                    # it word in capitalized -> return 2
                    capitalization_ids.extend([2]*len(word_ids))
                else:
                    # it word in lowercase -> return 1
                    capitalization_ids.extend([1]*len(word_ids))
            return {
                'input_ids': input_ids[:max_length], 
                'capitalization_ids': capitalization_ids,
                'token_type_ids': [0] * len(capitalization_ids),
                'attention_mask': [1] * len(capitalization_ids),
            }
```

Could someone help and give some advice to make the seed reasonable?

**Comment by maiiabocharova** [comment]
Thank you! Worked perfectly and now it's very fast!
